flang/runtime/execute.cpp

```diff
@@ -122,10 +122,22 @@ void RTNAME(ExecuteCommandLine)(const Descriptor &command, bool wait,
 
   if (exitstat) {
     RUNTIME_CHECK(terminator, IsValidIntDescriptor(exitstat));
+    auto exitstatKind{exitstat->type().GetCategoryAndKind()->second};
+    if (exitstatKind < 4) {
```

Comment:
The runtime code doesn't need to check these descriptors -- that will have been done statically when the intrinsic procedure reference was analyzed!

What has to be done in the runtime is using those descriptors to store the integer result values properly, and I think that the code already does that.

Reply:
Sorry I misunderstood your comment.